spring/src/main/java/io/cucumber/spring/SpringFactory.java

```diff
@@ -139,8 +139,10 @@ public void start() {
         notifyContextManagerAboutTestClassStarted();
         if (beanFactory == null || isNewContextCreated()) {
             beanFactory = testContextManager.getBeanFactory();
-            for (Class<?> stepClass : stepClasses) {
-                registerStepClassBeanDefinition(beanFactory, stepClass);
+            synchronized (beanFactory) {
```

Comment:
We also have a synchronization step a few lines up inside `CucumberTestContextManager`. For clarity we could synchronize all operations in the start method.

Reply:
done

Reply:
So now - we synchronize the start method statically. So for all instances of SpringFactory created they will start in serial